### PR TITLE
Part 4 - Adding Ticketmaster API to eventonica

### PR DESCRIPTION
**Pull request by nazreenpe**
- fetch events from ticketmaster for given keyword

### CODE REVIEW

**Comment by nazreenpe** [comment]
This PR has commits for part 3 as well, as it has not yet been accepted

File: eventonica-part-4/src/eventRecommender.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        );
+        let eventsByCategory = [];
+        eventKeys.forEach(key => eventsByCategory.push(this.events[key]));
+        return eventsByCategory;
+    }
+}
+
+class User {
+    constructor(fName, lName, password) {
+        this.fName = fName || "Anonymous";
```
Comment (datadaveshin) [inline eventonica-part-4/src/eventRecommender.js]: Not needed with your setup, but could add to index.html that the input for these are required, same for line 80.

File: eventonica-part-4/src/eventRecommender.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            key => this.events[key].category == category
+        );
+        let eventsByCategory = [];
+        eventKeys.forEach(key => eventsByCategory.push(this.events[key]));
+        return eventsByCategory;
+    }
+}
+
+class User {
+    constructor(fName, lName, password) {
```
Comment (datadaveshin) [inline eventonica-part-4/src/eventRecommender.js]: While no biggie here, using more explicit names can help. `fname` is often used in Python and `FName` in unreal engine for other purposes. Reviewers will generally know that `firstName` or for say Python `first_name` is probably the first name of someone.

File: eventonica-part-4/static/scripts/jQuery-scripts.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                let category = tmEvent.classifications[0].genre.name;
+                let date = moment(new Date(tmEvent.dates.start.dateTime)).format("MM/DD/YYYY");
+
+                let event = new Event(name, date, category);
+                eventRecommender.addEvent(event);
+
+                refreshEvents("tm-events-by-keyword");
+            },
+            error: function (xhr, status, err) {
+                console.log("Error fecthing data from Ticketmaster")
```
Comment (datadaveshin) [inline eventonica-part-4/static/scripts/jQuery-scripts.js]: Checkout [console.error()](https://developer.mozilla.org/en-US/docs/Web/API/Console/error) to see how you can make error messages more informative perhaps for the future

**Pull request merged**